library/core/src/num/dec2flt/common.rs

```diff
@@ -0,0 +1,220 @@
+//! Common utilities, for internal use only.
+
+use crate::ptr;
+
+/// Helper methods to process immutable bytes.
+pub trait ByteSlice: AsRef<[u8]> {
+    unsafe fn get_unchecked(&self, i: usize) -> u8 {
+        // SAFETY: same safety as get_unchecked
+        unsafe { *self.as_ref().get_unchecked(i) }
+    }
+
+    unsafe fn first_unchecked(&self) -> u8 {
+        debug_assert!(!self.as_ref().is_empty());
+        // SAFETY: same safety as first_unchecked, valid as long as self isn't empty.
```

Comment:
`same safety as first_unchecked` typo? Self refers in comment.

Reply:
@klensy My mistake, thanks for catching this.